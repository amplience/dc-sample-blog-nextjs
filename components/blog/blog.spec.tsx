/* eslint-env jest */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Blog from './blog';
import blogPostFixture from '../../tests/fixtures/single-blog-post-data-object.json';

describe('Blog', () => {
  test('renders full blog', async () => {
    const component = ShallowRenderer.createRenderer();
    component.render(<Blog blogPost={blogPostFixture} />);
    expect(component.getRenderOutput()).toMatchSnapshot();
  });
});
