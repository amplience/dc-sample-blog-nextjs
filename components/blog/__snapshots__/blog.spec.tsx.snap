// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog renders full blog 1`] = `
<React.Fragment>
  <div
    className="jsx-452798245 content-header"
  >
    <BlogPostAuthor
      authors={
        Array [
          Object {
            "_meta": Object {
              "deliveryId": "b1379f67-8934-49cb-94b3-58b679096c2c",
              "name": "blog-author",
              "schema": "https://schema.localhost.com/author.json",
            },
            "avatar": Object {
              "_meta": Object {
                "deliveryId": "9ef5a6d0-5a34-414f-9fd3-1098698c6629",
                "name": "blog-avatar",
                "schema": "https://schema.localhost.com/image.json",
              },
              "altText": "The dashing Sir Blog avatar",
              "image": Object {
                "_meta": Object {
                  "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link",
                },
                "defaultHost": "i1.adis.ws",
                "endpoint": "blogltd",
                "id": "f9549d3d-f616-487c-b1ad-fad5152f2c7d",
                "name": "man-business-suit",
              },
            },
            "name": "Sir Blog",
          },
        ]
      }
      date="August 07, 2019"
      readTime={12}
    />
    <TagChips
      tags={
        Array [
          "sir blog",
          "you're it",
          "No tiggy backs",
        ]
      }
    />
    <BlogPostHeroBanner
      subTitle="Some text for the description that explains this blog post"
      title="My First Blog Post"
    />
  </div>
  <div
    className="jsx-452798245 blog-image"
  >
    <Picture
      image={
        Object {
          "_meta": Object {
            "deliveryId": "4b44e870-bf71-43f3-81b3-c46c32f4cde5",
            "name": "bag-image",
            "schema": "https://schema.localhost.com/image.json",
          },
          "altText": "crotch bags",
          "image": Object {
            "_meta": Object {
              "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link",
            },
            "defaultHost": "i1.adis.ws",
            "endpoint": "blogltd",
            "id": "edc77f02-0e92-4b1a-88f9-711cacb5d650",
            "name": "friends-shopping-bags",
          },
        }
      }
      sources={
        Array [
          Object {
            "di": Object {
              "scaleFit": "poi",
              "sm": "c",
              "w": 2048,
            },
            "media": "(min-width: 1080px)",
          },
          Object {
            "di": Object {
              "scaleFit": "poi",
              "sm": "c",
              "w": 1080,
            },
            "media": "(min-width: 768px)",
          },
          Object {
            "di": Object {
              "scaleFit": "poi",
              "sm": "c",
              "w": 768,
            },
            "media": "(min-width: 480px)",
          },
          Object {
            "di": Object {
              "scaleFit": "poi",
              "sm": "c",
              "w": 480,
            },
          },
        ]
      }
    />
  </div>
  <div
    className="jsx-452798245 content-wrapper"
  >
    <Content
      content={
        Array [
          Object {
            "_meta": Object {
              "deliveryId": "a61f458e-7c22-4b5b-9b3d-962df3cbcdcd",
              "name": "casual-crotch-shot",
              "schema": "https://schema.localhost.com/image.json",
            },
            "altText": "Alt text",
            "image": Object {
              "_meta": Object {
                "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/image-link",
              },
              "defaultHost": "i1.adis.ws",
              "endpoint": "blogltd",
              "id": "e1b511d2-1a33-47e7-8dc7-f460534cb0c7",
              "name": "casual-wear",
            },
          },
          Object {
            "_meta": Object {
              "deliveryId": "f08432cd-525f-4e50-ab34-3316081f48bd",
              "name": "my-first-blog-text",
              "schema": "https://schema.localhost.com/text.json",
            },
            "text": "# This is my first Blog

## Look-a-header 2

### Look-a-header 3

#### Look-a-header 4

##### Look-a-header 5

###### Look-a-header 6

_**This**_ _is_ _some_ ~~sample~~ **text**.

![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png \\"Logo Title Text 1\\")

This is some more sample text with a [LINK](http://bbc.co.uk)

> Comments are cool!
> 
> See what I mean

Look some code:

\`\`\`javascript
const something = 'anything';
calls(something);

\`\`\`

inline \`code\` is awesome underlined

* * *

<h4>A title</h4>

A table too:

| Hello | world | table |
| --- | --- | --- |
| some | values | here |
| some | values | here |
| some | values | here |
| some | values | here |

Get listed

*   Item One
*   Item Two
*   Item Three",
          },
          Object {
            "_meta": Object {
              "deliveryId": "ad568b60-fd02-4302-9ccc-9e35dac24255",
              "name": "my-first-blog-video",
              "schema": "https://schema.localhost.com/video.json",
            },
            "srcSet": Array [],
            "video": Object {
              "_meta": Object {
                "schema": "http://bigcontent.io/cms/schema/v1/core#/definitions/video-link",
              },
              "defaultHost": "i1.adis.ws",
              "endpoint": "blogltd",
              "id": "721044de-d125-4a1a-8ddc-2201b9463f2d",
              "name": "SampleVideo_1280x720_5mb",
            },
          },
        ]
      }
    />
  </div>
  <JSXStyle
    id="452798245"
  >
    .content-header.jsx-452798245{margin:auto;max-width:740px;}.content-header.jsx-452798245 div.tag-chips{padding-top:25px;}.content-wrapper.jsx-452798245{margin:auto;max-width:675px;padding-bottom:100px;}.blog-image.jsx-452798245 img{object-fit:cover;width:100%;}footer{margin-top:100px;}@media (max-width:675px){.content-header.jsx-452798245,.content-footer.jsx-452798245,.content-wrapper.jsx-452798245{display:block;padding:0 20px;}}
  </JSXStyle>
</React.Fragment>
`;
